zerver/lib/test_classes.py

```diff
@@ -1077,7 +1077,7 @@ def assert_streaming_content(self, response: "TestHttpResponse", result: bytes)
         self.assertEqual(result, data)
 
     def assert_json_success(
-        self, result: Union["TestHttpResponse", HttpResponse]
+        self, result: Union["TestHttpResponse", HttpResponse], has_ignored_parameters: bool = False
```

Comment:
I think a better API might be `ignored_parameters=["first", "second"]` -- then validation of what ignored parameters appear can occur here, rather than in the caller.

(We'll probably want something similar for the JSON error code path as well).

Reply:
That makes sense. Updated for that change.